synth.py

```diff
@@ -2,61 +2,68 @@
 version = 'v1'
-library = gapic.node_library(
-    'pubsub', version, config_path="/google/pubsub/artman_pubsub.yaml")
+library = gapic.typescript_library(
+    'pubsub',
+    version,
+    generator_args={
+        'grpc-service-config': f'google/pubsub/{version}/pubsub_grpc_service_config.json',
+        'package-name': f'@google-cloud/pubsub',
+        'main-service': f'pubsub',
+        'bundle-config': f'google/pubsub/{version}/pubsub_gapic.yaml',
+        'template': f'typescript_gapic'
+    },
+    proto_path=f'/google/pubsub/{version}',
+    extra_proto_files=['google/cloud/common_resources.proto']
+)
 
 # skip index, protos, package.json, and README.md
 s.copy(
     library,
-    excludes=['package.json', 'README.md', 'src/index.js'])
+    excludes=['package.json', 'README.md', 'src/index.ts'])
 
 templates = common_templates.node_library(source_location='build/src')
 s.copy(templates)
 
-# https://github.com/googleapis/gapic-generator/issues/2127
-s.replace("src/v1/subscriber_client.js",
-          "  }\n\s*/\*\*\n\s+\* The DNS address for this API service\.",
-          "\n    // note: editing generated code\n"
-          "    this.waitForReady = function(deadline, callback) {\n"
-          "      return subscriberStub.then(\n"
-          "        stub => stub.waitForReady(deadline, callback),\n"
-          "        callback\n"
-          "      );\n"
-          "    };\n"
-          "    this.getSubscriberStub = function() {\n"
-          "      return subscriberStub;\n"
-          "    };\n"
-          "\g<0>")
+# surgery in client.ts file
+clients = ['publisher', 'subscriber']
+for client_name in clients:
+    client_file = f'src/v1/{client_name}_client.ts'
+    s.replace(client_file,
+              f'import \* as gapicConfig from \'\.\/{client_name}_client_config\.json\';',
+              f'import * as gapicConfig from \'./%s_client_config.json\';\nimport {{IamClient}} from \'../helper\';' % client_name,
+              )
+
+    s.replace(client_file,
+              'private \_terminated = false;',
+              'private _terminated = false; \n private _iamClient: IamClient;')
 
-# Update path discovery due to build/ dir and TypeScript conversion.
-s.replace("src/v1/publisher_client.js", "../../package.json", "../../../package.json")
-s.replace("src/v1/subscriber_client.js", "../../package.json", "../../../package.json")
+    s.replace(client_file,
+              '\/\/ Determine the client header string.',
+              'this._iamClient = new IamClient(opts); \n // Determine the client header string.')
 
-# [START fix-dead-link]
-s.replace('src/**/doc/google/protobuf/doc_timestamp.js',
-        'https:\/\/cloud\.google\.com[\s\*]*http:\/\/(.*)[\s\*]*\)',
-        r"https://\1)")
+    # fix tslint issue due to mismatch gts version with gapic-generator-typescript
+    # it should be removed once pubsub upgrade gts 2.0.0
+    s.replace(client_file, '\/\/ eslint\-disable\-next\-line\ \@typescript\-eslint\/no\-explicit\-any',
```

Comment:
Can we do both at the same time?

Reply:
Sorry, not understanding here, what are the both reference to? :)

Reply:
The `TODO` above says we need to upgrade to `gts@2.0.0`.  That happens here right?  Can we get rid of this synth change?

Reply:
I think Justin means the two `replace` items?
we can put them in an array and iterate the array to replace the text. But not sure there's a way to do it just in one line, as `s.replace(file_name, regex, replaceText)` and the regex is not the same.

Thanks!

Reply:
> The `TODO` above says we need to upgrade to `gts@2.0.0`. That happens here right? Can we get rid of this synth change?

@bcoe  @feywind and I had a discussion to isolate upgrade gts from typescript conversion. Pub/Sub is  last wave to dep node 8 and upgrade gts.

Reply:
Honestly this PR does some of the Node 8 deprecation already... it's nice to have stuff separated out, but it wouldn't be terrible to go ahead and add the deprecation pieces like gts here.

Reply:
I don't have a strong opinion here. For spanner, I'm combining gtsv2, TypeScript, and Node 8 deprecation, and this is something that makes it harder. Making it step by step here makes sense.